examples/next-prisma-starter/src/utils/trpc.ts

```diff
@@ -18,3 +18,15 @@ export const trpc = createReactQueryHooks<AppRouter>();
 export type inferQueryOutput<
   TRouteKey extends keyof AppRouter['_def']['queries'],
 > = inferProcedureOutput<AppRouter['_def']['queries'][TRouteKey]>;
+
+export type inferQueryInput<
+  TRouteKey extends keyof AppRouter['_def']['queries'],
+> = inferProcedureInput<AppRouter['_def']['queries'][TRouteKey]>;
+
+export type inferMutationOutput<
+  TRouteKey extends keyof AppRouter['_def']['mutations'],
+> = inferProcedureOutput<AppRouter['_def']['mutations'][TRouteKey]>;
+
+export type inferMutationInput<
+  TRouteKey extends keyof AppRouter['_def']['mutations'],
+> = inferProcedureInput<AppRouter['_def']['mutations'][TRouteKey]>;
```

Comment:
Maybe we should consider moving this logic to trpc. It could also help in case the signature of `AppRouter` has to change in the future.

Reply:
Agreed